### DISCUSSION TITLE
🌟 Message action button should be under content

### DISCUSSION THREAD

**Discussion by fraincs** [gsoft-inc/sg-orbit#1115] [Ideas]
## Description

Alignment of the Action button in a message should be under the main content. This will help maintain constitency between responsive view and standard view as well as look better when a Message is in a Modal with buttons.

## Checklist

- [ ] Figma
- [ ] Documentation
- [ ] States
- [ ] Design respect accessibility specs

**Comment by fraincs** [comment]
It has been brought to my attention that since we could use Message as a Toast down the line the alignment of the button suggested above could be problematic. If it is something still in the cards, maybe revamping the padding is a better option. As here everything seems misaligned.

![image](https://user-images.githubusercontent.com/361632/154072006-8ad42003-07d2-43c9-9322-3e72fb4d191a.png)

**Comment by patricklafrance** [comment]
What about cases were we want to display something more like a banner style? Meaning, a short message and an action, do we then introduce a new Banner component? It was a common use case back in the day.

I don't mind introducing a new Banner component for this use case if we still have it but we need to think about this.

I remember discussing with Gab that most of the actions in a Message should be "links" instead "buttons" because it usually redirect to another page.

Something similar to Paste

![image](https://user-images.githubusercontent.com/794579/154115742-372cdd1c-87c1-4ab6-a040-d9de98873cdd.png)

Buttons should only be used for "commands" not for "navigation".

**Comment by fraincs** [comment]
> I don't mind introducing a new Banner component

This also opens the door to introducing a Toast component that has a similar look but different layout.

**Comment by patricklafrance** [comment]
Well yes, everything is possible, not everything is a good idea ;)

We must be cautious about the multiplication of components. Adding a new component is the easies solution but also a patch to maybe a deep problem. Before doing so it's important to spend several hours (maybe days) analyzing the options. As the DS adoption grows it will becomes very hard to remove an existing component. We don't want to endup with a pile of deprecated components to maintains.

I was suggesting a Banner component because it's been an ongoing discussion for a while.

IMO, the current problem with Message might be misusage of action buttons and a lack of usage guidelines?

In Apricot, I think most of our current actions should be links embedded in the text because they have navigation purpose (I consider opening a Modal navigation).

I also think we should have a rule that only a single Message (without a title) should have an action.

Maybe it's too restrictive? As I reflect about this issue, I am pretty sure these are some of the guidelines we agreed on when we discussed the Message component during our sync but we didn't wrote anything down.

**Comment by patricklafrance** [comment]
Some thoughts... thinking out loud...

If we choose to move the button under the content then we can't have "one line" message anymore. I remember that back in the days it was an issue for Apricot. Is it still an issue?

- **If it's not an issue** then it's an acceptable solution. Does a Message can now have multiple actions? If yes, then the component design must be rethink. In either case, we will not be able to use it in a Toast anymore. Do we want Toast in the future? If yes, then are we willing to create and maintain a new component for this? Maybe it's a good idea, I don't know.

- **If it's an issue** then we have to introduce a new Banner component to support "one line" message.

We could end up with Message, Banner and Toast components (we also have Alert).

I guess it's quite easy to undertand when to use a Toast component. The difference between a Message and a Banner is less obvious though, it would have to be documented so the devs and designers choose the right component. It could be related to the positioning of the actions and the usage (or not) of a title.

Semantically though, "Message" doesn't make much sense anymore to me because Banner & Toast are also "messages". A new name might have to be found.

I guess the real questions is: Do we really have needs for all of this? I think back in the days Gab answers was no and that's why we stuck to "Message" but we didn't made the guidelines explicit :(

**Comment by patricklafrance** [comment]
Other options are still available like:

- Vertically center the action button and the close button
- Add more vertical padding to a Message

**Comment by kcharen** [comment]
@fraincs The topic of button placement has surfaced again. Wondering if we can prioritize and tackle this subject now?

**Comment by patricklafrance** [comment]
@fraincs is the rebrand adressing this issue?
